plugins/templates/index.js

```diff
@@ -16,13 +18,62 @@ const removeTagRoute = require('./removeTag');
  * @param  {Function} next              Function to call when done
  */
 exports.register = (server, options, next) => {
+    /**
+     * Throws error if a credential does not have permission to remove template
+     * If credential has access, resolves to true
+     * @method canRemove
+     * @param {Object}  credentials              Credential object from Hapi
+     * @param {String}  credentials.username     Username of the person logged in (or build ID)
+     * @param {String}  credentials.scmContext   Scm of the person logged in (or build ID)
+     * @param {Array}   credentials.scope        Scope of the credential (user, build, admin)
+     * @param {String}  [credentials.pipelineId] If credential is a build, this is the pipeline ID
+     * @param {Object}  template                 Target template object
+     * @param {String}  permission               Required permission level
+     * @return {Promise}
+     */
+    server.expose('canRemove', (credentials, template, permission) => {
+        const { username, scmContext, scope } = credentials;
+        const { userFactory, pipelineFactory } = server.root.app;
+        const forbiddenMsg = `User ${username} does not have ` +
```

Comment:
You use this same error message for `build` scope (line 62), so the message will be kinda weird since it's not a `User`